core/src/main/java/com/sequenceiq/cloudbreak/controller/v4/EnvironmentV4Controller.java

```diff
@@ -11,71 +11,66 @@
+import com.sequenceiq.environment.api.v1.environment.model.response.SimpleEnvironmentResponses;
 
 @Controller
 @Transactional(TxType.NEVER)
 public class EnvironmentV4Controller implements EnvironmentV4Endpoint {
 
     @Inject
-    private EnvironmentService environmentService;
+    private DatalakePrerequisiteService datalakePrerequisiteService;
 
     @Inject
-    private DatalakePrerequisiteService datalakePrerequisiteService;
+    private EnvironmentClientService environmentClientService;
 
     @Override
-    public DetailedEnvironmentV4Response post(Long workspaceId, @Valid EnvironmentV4Request request) {
-        return environmentService.createForLoggedInUser(request, workspaceId);
+    public DetailedEnvironmentResponse post(Long workspaceId, @Valid EnvironmentRequest request) {
+        return environmentClientService.create(request);
     }
 
     @Override
-    public SimpleEnvironmentV4Responses list(Long workspaceId) {
-        return new SimpleEnvironmentV4Responses(environmentService.listByWorkspaceId(workspaceId));
+    public SimpleEnvironmentResponses list(Long workspaceId) {
+        return environmentClientService.list();
     }
 
     @Override
-    public DetailedEnvironmentV4Response get(Long workspaceId, String environmentName) {
-        return environmentService.get(environmentName, workspaceId);
+    public DetailedEnvironmentResponse get(Long workspaceId, String environmentCrn) {
+        return environmentClientService.get(environmentCrn);
     }
 
     @Override
-    public SimpleEnvironmentV4Response delete(Long workspaceId, String environmentName) {
-        return environmentService.delete(environmentName, workspaceId);
+    public SimpleEnvironmentResponse delete(Long workspaceId, String environmentCrn) {
+        return environmentClientService.delete(environmentCrn);
     }
 
     @Override
-    public DetailedEnvironmentV4Response edit(Long workspaceId, String environmentName, @NotNull EnvironmentEditV4Request request) {
-        return environmentService.edit(workspaceId, environmentName, request);
+    public DetailedEnvironmentResponse edit(Long workspaceId, String environmentCrn, @NotNull EnvironmentEditRequest request) {
+        return environmentClientService.edit(environmentCrn, request);
     }
 
     @Override
-    public DetailedEnvironmentV4Response attach(Long workspaceId, String environmentName, @Valid EnvironmentAttachV4Request request) {
+    public DetailedEnvironmentV4Response attach(Long workspaceId, String environmentCrn, @Valid EnvironmentAttachV4Request request) {
         throw new UnsupportedOperationException("Attaching resource to an environment is not supported anymore!");
     }
 
     @Override
-    public DetailedEnvironmentV4Response detach(Long workspaceId, String environmentName, @Valid EnvironmentDetachV4Request request) {
+    public DetailedEnvironmentV4Response detach(Long workspaceId, String environmentCrn, @Valid EnvironmentDetachV4Request request) {
```

Comment:
Shouldn't it be DetailedEnvironmentResponse ?

Reply:
Nope, this is the old Endpoint in cloudbreak which I left here to make it easier to rewire integration tests to use the new env service. If I had deleted this class, most of the integration tests should be deleted too. And it wouldn't be much fun rewriting them.